packages/export-selector-ui/src/lib/components/ExportAccount.tsx

```diff
@@ -0,0 +1,351 @@
+import React, { Fragment, useEffect, useState } from "react";
+import BN from "bn.js";
+import type {
+  BrowserWalletBehaviour,
+  InjectedWalletBehaviour,
+  ModuleState,
+  Wallet,
+  WalletSelector,
+  AccountImportData,
+} from "@near-wallet-selector/core";
+import { encodeAccountData } from "@near-wallet-selector/core";
+
+import { utils, providers } from "near-api-js";
+import type {
+  FunctionCallPermissionView,
+  AccessKeyView,
+  AccountView,
+} from "near-api-js/lib/providers/provider";
+
+import { AccountSelect } from "./AccountSelect";
+import { Passphrase } from "./Passphrase";
+import { NoInterface } from "./NoInterface";
+import { Complete } from "./Complete";
+
+interface ExportAccountProps {
+  alertMessage: string | null;
+  module?: ModuleState;
+  onCloseModal: () => void;
+  onWarning: () => void;
+  onBack: () => void;
+  accounts: Array<AccountImportData>;
+  selector: WalletSelector;
+  wallet: ModuleState<Wallet>;
+  onComplete?: (accounts: Array<string>) => void;
+}
+
+const EXPORT_ACCOUNT_STEPS = {
+  ACCOUNT_SELECTION: "ACCOUNT_SELECTION",
+  GET_PASSPHRASE: "GET_PASSPHRASE",
+  NO_INTERFACE: "NO_INTERFACE",
+  COMPLETE: "COMPLETE",
+};
+
+const ACCESS_KEY_TYPES = {
+  LEDGER: "Ledger",
+  FULL_ACCESS_KEY: "Full Access Key",
+  MULTI_SIG: "Multi-Sig",
+  UNKNOWN: "Unknown",
+};
+
+const permissionToType = (
+  permission: string | FunctionCallPermissionView
+): string => {
+  if (permission === "FullAccess") {
+    return ACCESS_KEY_TYPES.FULL_ACCESS_KEY;
+  }
+  //@ts-ignore
+  if (permission?.FunctionCall?.method_names.includes("__wallet__metadata")) {
+    return ACCESS_KEY_TYPES.LEDGER;
+  }
+  const multiSigMethods = [
+    "add_request",
+    "add_request_and_confirm",
+    "delete_request",
+    "confirm",
+  ];
+  if (
+    //@ts-ignore
+    permission?.FunctionCall?.method_names.every((method: string) =>
+      multiSigMethods.includes(method)
+    )
+  ) {
+    return ACCESS_KEY_TYPES.MULTI_SIG;
+  }
+  return ACCESS_KEY_TYPES.UNKNOWN;
+};
+
+interface getAccountTypeProps {
+  provider: providers.Provider;
+  accountId: string;
+  publicKey: string;
+}
+
+const getAccountType = async ({
+  provider,
+  accountId,
+  publicKey,
+}: getAccountTypeProps) => {
+  const result = await provider
```

Comment:
💄 I think this should either return the promise chain or use a `try`/`catch` with `await`, mixing is a little awkward when the result of the chain is all that needs to be returned, e.g.

```js
return provider.query(...);
```

or

```js
try {
  const { permission } = await provider.query(...);
  return permissionToType(permission);
} catch {
  return ...
}
```